base/errorshow.jl

```diff
@@ -1022,6 +1022,18 @@ end
 
 Experimental.register_error_hint(noncallable_number_hint_handler, MethodError)
 
+# handler for displaying a hint in case the user tries to setindex
+# the instance of a number (probably attempting to use wrong indexing)
+# eg: a = [1 2; 3 4]; a[1][2] = 5
+function nonsetable_number_hint_handler(io, ex, arg_types, kwargs)
+    if ex.f == setindex! && arg_types[1] <: Number
+        print(io, "\nAre you trying to index into an array? Separate each index with commas: ")
+        printstyled(io, "a[1, 2]", color=:cyan)
```

Comment:
"Separate each index" seems like it needs a plural. 

And maybe it's worth showing the wrong form too, so that you know what to look for if it's a big block of code.
```suggestion
        print(io, "\nAre you trying to index into an array? For multi-dimensional arrays, separate the indices with commas: ")
        printstyled(io, "a[1, 2]", color=:cyan)
        print(io, " rather than a[1][2]")
```